### PR TITLE
bpo-35293: Remove RemovedInSphinx40Warning

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue35293

### CODE REVIEW

**Comment by corona10** [comment]
The current failure is not related to this PR
See: https://bugs.python.org/issue41762
```
/Users/corona10/oss/cpython/Doc/library/string.rst:311:duplicate token description of sign, other instance in library/functions
```

**Comment by vstinner** [comment]
> The current failure is not related to this PR

I don't understand. All jobs are now green. Did you re-run some jobs?

**Comment by corona10** [comment]
> I don't understand. All jobs are now green.

I expected that `Travis CI - Pull Request` will run but not on this PR.

**Comment by corona10** [comment]
@vstinner @njsmith Please take a look

**Comment by vstinner** [comment]
I fixed https://bugs.python.org/issue41762 with commit 8af239eacfcf52e4e0e2b0223e7cea4672309483, PR #22281.

**Comment by corona10** [comment]
> I fixed https://bugs.python.org/issue41762 with commit 8af239e, PR #22281.

Nice!, I rebased it!

File: Misc/NEWS.d/next/Documentation/2020-09-12-17-37-13.bpo-35293._cOwPD.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Remove RemovedInSphinx40Warning when building sphinx. Patch by Dong-hee Na.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Documentation/2020-09-12-17-37-13.bpo-35293._cOwPD.rst]:
```suggestion
Fix RemovedInSphinx40Warning when building the documentation. Patch by Dong-hee Na.
```

File: Doc/tools/extensions/pyspecific.py [hunk 2]
```diff
@@ -31,7 +31,16 @@
...
+
+try:
+    from sphinx.domains.python import PyMethod
+except ImportError:
+    from sphinx.domains.python import PyClassmember as PyMethod
+
+try:
+    from sphinx.domains.python import PyFunction
+except ImportError:
+    from sphinx.domains.python import PyModulelevel as PyFunction
```
Comment (vstinner) [inline Doc/tools/extensions/pyspecific.py:39]: Would it make sense to merge the two try/except? I expect that PyMethod and PyFunction were added to the same Sphinx version.

**Comment by vstinner** [comment]
I confirm that the PR fix RemovedInSphinx40Warning warnings.

Without the PR:

```
$ cd Doc
$ make clean
$ make venv PYTHON=../python  && make html SPHINXOPTS="-j10"
(...)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:310: RemovedInSphinx40Warning: PyClassmember is deprecated. Please check the implementation of <class 'pyspecific.PyCoroutineMethod'>
  return PyClassmember.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:304: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyCoroutineFunction'>
  return PyModulelevel.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:278: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyDecoratorFunction'>
  return PyModulelevel.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:316: RemovedInSphinx40Warning: PyClassmember is deprecated. Please check the implementation of <class 'pyspecific.PyAwaitableFunction'>
  return PyClassmember.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:278: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyDecoratorFunction'>
  return PyModulelevel.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:278: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyDecoratorFunction'>
  return PyModulelevel.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:310: RemovedInSphinx40Warning: PyClassmember is deprecated. Please check the implementation of <class 'pyspecific.PyCoroutineMethod'>
  return PyClassmember.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:278: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyDecoratorFunction'>
  return PyModulelevel.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:322: RemovedInSphinx40Warning: PyClassmember is deprecated. Please check the implementation of <class 'pyspecific.PyAwaitableMethod'>
  return PyClassmember.run(self)
/home/vstinner/python/master/Doc/tools/extensions/pyspecific.py:278: RemovedInSphinx40Warning: PyModulelevel is deprecated. Please check the implementation of <class 'pyspecific.PyDecoratorFunction'>
  return PyModulelevel.run(self)
```

With the PR, I don't see these warnings anymore:

```
$ cd Doc
$ make clean
$ make venv PYTHON=../python  && make html SPHINXOPTS="-j10"
(...)
```

**Review by vstinner — APPROVED** [review]
LGTM.

**Comment by corona10** [comment]
@vstinner Thank you Victor!

**Pull request merged**